### DISCUSSION TITLE
how to use Obx() out side the widget tree?

### DISCUSSION THREAD

**Discussion by praburocking** [jonataslaw/getx#1304] [General]
i am working on a flutter web project where, i have a variable in controller islogedIn, if it is true i need to display another page, if false i need to display another page. I am trying to implement something like this 

` Obx(() => authController.isLogedIn.value ? Get.toNamed('/home') : Get.toNamed('/login'));
`

but it is not working since it has to return a widget, so I also tried something like this

 ```
Obx(() { 
authController.isLogedIn.value ? Get.toNamed('/home') : Get.toNamed('/login')
return Container();
});

```
still, this code is not executed, Obx is getting executed only when it is a part of the widget tree. do we have any option to Observe a controller outside a widget tree ?

**Comment by eduardoflorence** [comment]
You do not need Obx for this use case, as you only need the current value when deciding where to navigate. You can only do this:
```dart
authController.isLogedIn.value ? Get.toNamed('/home') : Get.toNamed('/login'));
```

**Reply by praburocking** [reply]
this is my controller class


```
import 'package:get/get.dart';
import 'package:shared_preferences/shared_preferences.dart';

class AuthController extends GetxController {
  var isLogedIn = false.obs;
  var token = ''.obs;

  @override
  onInit() async {
    super.onInit();
    SharedPreferences prefs = await SharedPreferences.getInstance();
    await prefs.setString('token', 'qwertyuiop');
    token.value = prefs.getString('token');
    print(token.value);
    if (token.value != '' && token.value != null) {
      isLogedIn.value = true;
      print('isLogedIn value ==>' + isLogedIn.value.toString());
    }
  }

  updateLogedIn() {}
}

```


 and this is my widget in which I use the above controller: 



```
import 'package:flutter/material.dart';
import 'package:get/get.dart';
import 'package:myapp/welcome/controllers/AuthController.dart';
import 'package:myapp/welcome/views/login.dart';
import 'package:carousel_slider/carousel_slider.dart';

class WelcomePage extends StatelessWidget {
  final Function loginAction;
  WelcomePage({this.loginAction});

  @override
  Widget build(BuildContext context) {
    AuthController authController = Get.put(AuthController());
authController.isLogedIn.value ? Get.toNamed('/home'): Get.toNamed('/login'))
    print(authController.isLogedIn.value);

    Widget welcome = Container(
      width: MediaQuery.of(context).size.width,
      height: MediaQuery.of(context).size.height,
      decoration: BoxDecoration(color: Colors.blueGrey),
      child: Stack(
        children: [
          Container(
            height: MediaQuery.of(context).size.height,
            width: MediaQuery.of(context).size.width * 0.61,
            child: CarouselSlider(
              options: CarouselOptions(
                height: MediaQuery.of(context).size.height,
                viewportFraction: 1.0,
                enlargeCenterPage: false,
                // autoPlay: false,
              ),
              items: [
                "https://image.shutterstock.com/image-photo/mountains-under-mist-morning-amazing-260nw-1725825019.jpg",
              ].map((i) {
                return Builder(
                  builder: (BuildContext context) {
                    return Container(
                      child: Center(
                          child: Image.network(
                        i,
                        fit: BoxFit.cover,
                        height: MediaQuery.of(context).size.height,
                      )),
                    );
                  },
                );
              }).toList(),
            ),
          ),
          Login()
        ],
      ),
    );

    return welcome;
  }
}
```

So the `isLoggedIn` value changes to true only after the async init() of the controller  completes, before that the value of `isLoggedIn` will be false only.  thats is why i need to observe the variable to make page switch.

**Comment by eduardoflorence** [comment]
And where is the part of the code that uses `authController.isLogedIn.value ? Get.toNamed('/home'): Get.toNamed('/login')) `?

**Reply by praburocking** [reply]
`authController.isLogedIn.value ? Get.toNamed('/home'): Get.toNamed('/login'))` will be there inside the WelcomePage widget itself, i have updated the class in the previous comment.

**Reply by eduardoflorence** [reply]
Show code (WelcomePage)

**Reply by praburocking** [reply]
hi, both authController and welcomPage code is present in the above comment, if you need more insight about the project and its structure to suggest a solution, kindly have a look at [this](https://github.com/praburocking/flutterWebTraining) , here I am trying to use flutter and getx for creating web app, and I am having issue in the part where, when the user comes to the site i need to check whether he is already signed in (with authtoken store in localstorage), if he is already signed in will redirect him to '/home' , if not will redirect him to '/login'.
